src/cpuinfo_x86.c

```diff
@@ -147,6 +147,249 @@ static X86CacheLevelInfo MakeX86CacheLevelInfo(int level,
+    case 0x86:
+      return MakeX86CacheLevelInfo(2, DATA_CACHE, 512, 4, 32, -1, 0);
+    case 0x87:
+      return MakeX86CacheLevelInfo(2, DATA_CACHE, 1 * 1024, 8, 64, -1, 0);
+    case 0xA0:
+      return MakeX86CacheLevelInfo(-1, DTLB, 4, 0xFF, -1, 32, 0);
+    case 0xB0:
+      return MakeX86CacheLevelInfo(-1, TLB, 4, 4, -1, 128, 0);
+    case 0xB1:
+      return MakeX86CacheLevelInfo(-1, TLB, 2 * 1024, 4, -1, 8, 0);
+    case 0xB2:
+      return MakeX86CacheLevelInfo(-1, TLB, 4, 4, -1, 64, 0);
+    case 0xB3:
+      return MakeX86CacheLevelInfo(-1, TLB, 4, 4, -1, 128, 0);
+    case 0xB4:
+      return MakeX86CacheLevelInfo(-1, TLB, 4, 4, -1, 256, 0);
+    case 0xB5:
+      return MakeX86CacheLevelInfo(-1, TLB, 4, 8, -1, 64, 0);
+    case 0xB6:
+      return MakeX86CacheLevelInfo(-1, TLB, 4, 8, -1, 128, 0);
+    case 0xBA:
+      return MakeX86CacheLevelInfo(-1, TLB, 4, 4, -1, 64, 0);
+    case 0xC0:
+      return MakeX86CacheLevelInfo(-1, TLB, 4, 4, -1, 8, 0);
+    case 0xC1:
+      return MakeX86CacheLevelInfo(-1, STLB, 4, 8, -1, 1024, 0);
+    case 0xC2:
+      return MakeX86CacheLevelInfo(-1, DTLB, 4, 4, -1, 16, 0);
+    case 0xC3:
+      return MakeX86CacheLevelInfo(-1, STLB, 4, 6, -1, 1536, 0);
+    case 0xCA:
+      return MakeX86CacheLevelInfo(-1, STLB, 4, 4, -1, 512, 0);
+    case 0xD0:
+      return MakeX86CacheLevelInfo(3, DATA_CACHE, 512, 4, 64, -1, 0);
+    case 0xD1:
+      return MakeX86CacheLevelInfo(3, DATA_CACHE, 1 * 1024, 4, 64, -1, 0);
+    case 0xD2:
+      return MakeX86CacheLevelInfo(3, DATA_CACHE, 2 * 1024, 4, 64, -1, 0);
+    case 0xD6:
+      return MakeX86CacheLevelInfo(3, DATA_CACHE, 1 * 1024, 8, 64, -1, 0);
+    case 0xD7:
+      return MakeX86CacheLevelInfo(3, DATA_CACHE, 2 * 1024, 8, 64, -1, 0);
+    case 0xD8:
+      return MakeX86CacheLevelInfo(3, DATA_CACHE, 4 * 1024, 8, 64, -1, 0);
+    case 0xDC:
+      return MakeX86CacheLevelInfo(3, DATA_CACHE, 1 * 1536, 12, 64, -1, 0);
+    case 0xDD:
+      return MakeX86CacheLevelInfo(3, DATA_CACHE, 3 * 1024, 12, 64, -1, 0);
+    case 0xDE:
+      return MakeX86CacheLevelInfo(3, DATA_CACHE, 6 * 1024, 12, 64, -1, 0);
+    case 0xE2:
+      return MakeX86CacheLevelInfo(3, DATA_CACHE, 2 * 1024, 16, 64, -1, 0);
+    case 0xE3:
+      return MakeX86CacheLevelInfo(3, DATA_CACHE, 4 * 1024, 16, 64, -1, 0);
+    case 0xE4:
+      return MakeX86CacheLevelInfo(3, DATA_CACHE, 8 * 1024, 16, 64, -1, 0);
+    case 0xEA:
+      return MakeX86CacheLevelInfo(3, DATA_CACHE, 12 * 1024, 24, 64, -1, 0);
+    case 0xEB:
+      return MakeX86CacheLevelInfo(3, DATA_CACHE, 18 * 1024, 24, 64, -1, 0);
+    case 0xEC:
+      return MakeX86CacheLevelInfo(3, DATA_CACHE, 24 * 1024, 24, 64, -1, 0);
+    case 0xF0:
+      return MakeX86CacheLevelInfo(-1, PREFETCH, 64, -1, -1, -1, 0);
+    case 0xF1:
+      return MakeX86CacheLevelInfo(-1, PREFETCH, 128, -1, -1, -1, 0);
+    case 0xFF:
+      return MakeX86CacheLevelInfo(-1, EMPTY, -1, -1, -1, -1, 0);
+  }
```

Comment:
wouldn't it be possible to make a table of structs out of it that hosts the parameters?

Reply:
> wouldn't it be possible to make a table of structs out of it that hosts the parameters?

I fail to see how a table of structs makes it simpler. Once you have the table you still need to index it.
Did I miss something?

Reply:
I'm not sure how it can be simplified. If you have any suggestions, I can implement them.